run-tests.php

```diff
@@ -2160,22 +2160,28 @@ function run_test($php, $file, $env)
 				return 'SKIPPED';
 			}
 
-			if (!strncasecmp('info', ltrim($output), 4)) {
-				if (preg_match('/^\s*info\s*(.+)\s*/i', $output, $m)) {
-					$info = " (info: $m[1])";
-				}
-			}
-
-			if (!strncasecmp('warn', ltrim($output), 4)) {
-				if (preg_match('/^\s*warn\s*(.+)\s*/i', $output, $m)) {
-					$warn = true; /* only if there is a reason */
-					$info = " (warn: $m[1])";
-				}
-			}
-
-			if (!strncasecmp('xfail', ltrim($output), 5)) {
+			if (!strncasecmp('info', ltrim($output), 4) && preg_match('/^\s*info\s*(.+)\s*/i', $output, $m)) {
```

Comment:
As a bit of drive-by cleanup, maybe do a `$output = trim($output)` before all this code? Then we can drop all those ltrims and the leading `\s*` from the regular expressions.

Reply:
done